src/Tagger/Database.hs

```diff
@@ -0,0 +1,49 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Tagger.Database
+  ( Config (..),
+    Handle,
+    parseConfig,
+    new,
+    close,
+    withHandle,
+    runQuery,
+  )
+where
+
+import Api.Config (dbConfig)
+import qualified Api.Config as AppConfig
+import Data.ByteString.Char8 (ByteString, unpack)
+import Data.Maybe (fromMaybe)
+import Hasql.Connection (Connection, acquire)
+import Hasql.Session (QueryError, Session, run)
+
+newtype Config = Config
+  { connectionString :: ByteString
+  }
+
+newtype Handle = Handle
+  { dbConnection :: Connection
+  }
+
+new :: Config -> IO Handle
+new config = do
+  eConn <- acquire . connectionString $ config
+  either
+    (fail . unpack . fromMaybe "unable to connect to the database")
+    (pure . Handle)
+    eConn
+
+parseConfig :: AppConfig.Config -> Config
+parseConfig =
+  Config . (AppConfig.connectionString . dbConfig)
+
+close :: Handle -> IO ()
```

Comment:
these two functions are not used, do we really need them?

Reply:
The Handle Pattern recommends including at least the `close` function even if it does nothing, as a way to have a standard "interface" for working with handles. It was my mistake when splitting the PR to not include the latest version of this module that is not `undefined` but that in turn does nothing.

I re-added the implementation of `close` and removed `withHandle`.